service/matching/matching_engine.go

```
@@ -2221,8 +2222,24 @@ func (e *matchingEngineImpl) DispatchNexusTask(ctx context.Context, request *mat
 	}
 
 	taskID := uuid.New()
+
+	namespaceID := namespace.ID(request.GetNamespaceId())
+	ns, err := e.namespaceRegistry.GetNamespaceByID(namespaceID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Buffer the deadline so we can still respond with timeout if we hit the deadline while dispatching
+	ctx, cancel := contextutil.WithDeadlineBuffer(ctx, matching.DefaultTimeout, e.config.MinDispatchTaskTimeout(ns.Name().String()))
+	defer cancel()
+
 	resp, err := pm.DispatchNexusTask(ctx, taskID, request)
 
+	// If we get a context error, that means we didn't manage to dispatch the task to a remote host
+	if ctx.Err() != nil {
+		return nil, ctx.Err()
+	}
+
```

You can just return the error here, not need for this special case.

```suggestion
```